Public header includes private header

While packaging cmark-gfm 0.29.0.gfm.2 libraries for build2 toolchain, I have noticed that the public header `cmark-gfm-core-extensions.h` includes the private header `config.h`:

```
#include "config.h" // for bool
```
Thus building a program that includes `cmark-gfm-core-extensions.h` against the installed cmark-gfm libraries fails with:

```
/tmp/install/include/cmark-gfm-core-extensions.h:10:10: fatal error: config.h: No such file or directory
```

--- a/extensions/cmark-gfm-core-extensions.h
+++ b/extensions/cmark-gfm-core-extensions.h
@@ -7,7 +7,7 @@ extern "C" {
 
 #include "cmark-gfm-extension_api.h"
 #include "cmark-gfm-extensions_export.h"
-#include "config.h" // for bool
+#include <stdbool.h>
 #include <stdint.h>
 
 CMARK_GFM_EXTENSIONS_EXPORT

